src/main/java/org/wildfly/security/x500/X500.java

```diff
@@ -137,4 +137,32 @@ private X500() {}
             return Arrays.copyOf(certificates, certificates.length, X509Certificate[].class);
         }
     }
+
+    /**
+     * Convert an X.500 attribute name (AVA keyword) to OID
+     * @param attribute X.500 attribute name
+     * @return corresponding OID or {@code null} if was not recognized
```

Comment:
1) but some of attributes handled by this method are also not applicable for DN. From that fact I supposed it is method for general usage. If it will be used only for DN, it is OK then.

I also suppose it will be applied here:
`
            <xs:extension base="principalDecoderType">
                <xs:attribute name="oid" type="xs:string" use="required">
                    <xs:annotation>
                        <xs:documentation>
                            The oid of the attribute to map.
                        </xs:documentation>
                    </xs:annotation>
                </xs:attribute>
`
Then rename from "oid" to "attribute" is OK. But in this example, I think not only attributes applicable in DN can be used.